cyclops/query/interface.py

```diff
@@ -139,20 +162,38 @@ class QueryInterfaceProcessed:
         The query.
     process_fn: Callable
         Process function to apply on the pandas dataframe returned from the query.
+    join: cyclops.query.ops.JoinArgs, optional
+        Join arguments to join the query with another table.
+    ops: cyclops.query.ops.Sequential, optional
+        Operations to perform on the query.
+    _data: pandas.DataFrame or dask.DataFrame
+        Data returned from executing the query, as Pandas DataFrame.
     _run_args: dict
         Private dictionary attribute to keep track of arguments
         passed to run() method.
-    data: pandas.DataFrame or dask.DataFrame
-        Data returned from executing the query, as Pandas DataFrame.
+
+    Notes
+    -----
+    After initialization, the query, join operation and chaining of provided operations,
+    are automatically done and the query attribute is updated.
 
     """
 
     database: Database
     _query: TableTypes
     process_fn: Callable
-    data: Optional[Union[pd.DataFrame, dd.DataFrame, None]] = None
+    join: Optional[qo.JoinArgs] = None
+    ops: Optional[qo.Sequential] = None
+    _data: Optional[Union[pd.DataFrame, dd.DataFrame, None]] = None
     _run_args: Dict = field(default_factory=dict)
 
+    def __post_init__(self) -> None:
+        """Post init method to chain operations with original query."""
+        if self.join:
+            self._query = qo.Join(**asdict(self.join))(self._query)
+        if self.ops:
+            self._query = self.ops(self._query)
```

Comment:
Are you intentionally checking for truthy-ness of `join` and `ops` or is your intention only to check for `None`? It it's the latter, I'd propose the explicit version:

```suggestion
        if self.join is not None:
            self._query = qo.Join(**asdict(self.join))(self._query)
        if self.ops is not None:
            self._query = self.ops(self._query)
```